docs/source/developers/index.rst

```diff
@@ -0,0 +1,43 @@
+.. _developers_guide:
+
+Developer's Guide
+==================
+
+This section contains information for developers who want to contribute to the cuML project.
+
+Deprecation Policy
+------------------
+
+cuML follows the policy of deprecating code for one release prior to removal. This applies
+to publicly accessible functions, classes, methods, attributes and parameters. During the
+deprecation cycle the old name or value is still supported, but will raise a deprecation
+warning when it is used.
+
+Code in cuML should not use deprecated cuML APIs.
+
+
+.. code-block:: python
```

Comment:
Should we show an example for libcuml as well?

Reply:
Sounds like a good idea, but what is `libcuml`? I feel like I should know the answer but I don't

Reply:
Here are some references from RMM as well. @csadorf what you linked seems to me acceptable as precedent set by other RAPIDS libraries https://github.com/rapidsai/rmm/pull/1690/files#diff-db547576924fdcc827a79d62925a34ea52e3e81f51908f1cee4b606d006f6ab8R115-R117